### DISCUSSION TITLE
Convert Entire Font to an Image Set.

### DISCUSSION THREAD

**Discussion by FullBleed** [ImageMagick/ImageMagick#5012] [Help]
On Windows using latest version of ImageMagick.

Is there some way to have every character in a selected font be rendered into a png or webp?

Specifically, black text on a transparent background.

Thanks.

**Comment by snibgo** [comment]
Create a text file of the characters you want. For example, alphabet.txt:
```
 !"#$%&'()*+,-./0123456789:;<=>?
@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_
`abcdefghijklmnopqrstuvwxyz{|}~
€‚ƒ„…†‡ˆ‰Š‹ŒŽ''""•–—˜™š›œžŸ
 ¡¢£¤¥¦§¨©ª«¬­®¯°±²³´µ¶·¸¹º»¼½¾¿
ÀÁÂÃÄÅÆÇÈÉÊËÌÍÎÏÐÑÒÓÔÕÖ×ØÙÚÛÜÝÞß
àáâãäåæçèéêëìíîïðñòóôõö÷øùúûüýþÿ
```
Then use a magick command:
```
magick -font Courier-New label:@\prose\alphabet.txt x.png
```


![x](https://user-images.githubusercontent.com/33812368/161566318-19ffbc60-3c22-4623-8708-aa8cb5e1d5be.png)

**Reply by FullBleed** [reply]
Unfortunately, the font I want to convert has over 1000 characters in it... so I'm not sure how to make a text file with every character in it... will have to see if what fmw42 says helps with that...

**Comment by fmw42** [comment]
See 

https://legacy.imagemagick.org/Usage/scripts/graphics_utf
https://legacy.imagemagick.org/Usage/scripts/show_fonts

**Comment by snibgo** [comment]
A font file may contain millions of glyphs. It may have codepoints (the "number" of the character) that have no glyph, as in the example I showed.

IM has no facility to count the number of glyphs in a font, nor to determine which codepoints have glyphs, other than rasterizing each possible codepoint and examining the resulting raster image. I suspect specialist font tools can do that.
